### PR TITLE
feat(compute): add compute consistency group remove disk

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description
Please review this PR first - https://github.com/GoogleCloudPlatform/java-docs-samples/pull/9681
Documentation - https://cloud.google.com/compute/docs/disks/async-pd/manage-consistency-groups?authuser=0#rest_1

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/CreateDiskConsistencyGroup.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+              .setDiskConsistencyGroupPolicy(
+                  ResourcePolicy.newBuilder().getDiskConsistencyGroupPolicy())
+              .build();
+
+      Operation response =
+          regionResourcePoliciesClient.insertAsync(project, region, resourcePolicy).get();
+
+      if (response.hasError()) {
+        return null;
+      }
```
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/CreateDiskConsistencyGroup.java:62]: ditto. throw informative exception instead.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      throws IOException, ExecutionException, InterruptedException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (ResourcePoliciesClient regionResourcePoliciesClient = ResourcePoliciesClient.create()) {
+      Operation response = regionResourcePoliciesClient
+          .deleteAsync(project, region, consistencyGroupName).get();
+
+      if (response.hasError()) {
+        return;
+      }
```
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java:52]: ditto, throw exception instead

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      System.out.println(
+          "Disk consistency group resource policy deleted successfully: "
+              + consistencyGroupName);
+    }
```
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java:56]: no need for this.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/RemoveDiskFromConsistencyGroup.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    //        RemoveResourcePoliciesDiskRequest request =
+    //            RemoveResourcePoliciesDiskRequest.newBuilder()
+    //                .setDisk(diskName)
+    //                .setDisksRemoveResourcePoliciesRequestResource(
+    //                    DisksRemoveResourcePoliciesRequest.newBuilder()
+    //                        .addAllResourcePolicies(Arrays.asList(consistencyGroupUrl))
+    //                        .build())
+    //                .setProject(project)
+    //                .setZone(location)
+    //                .build();
```
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/RemoveDiskFromConsistencyGroup.java:71]: nit: consider creating a separate sample for zonal disk

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/RemoveDiskFromConsistencyGroup.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+              .setRegionDisksRemoveResourcePoliciesRequestResource(
+                  RegionDisksRemoveResourcePoliciesRequest.newBuilder()
+                      .addAllResourcePolicies(Arrays.asList(consistencyGroupUrl))
+                      .build())
+              .build();
+
+      Operation response = disksClient.removeResourcePoliciesAsync(disksRequest).get();
+      if (response.hasError()) {
+        return null;
+      }
```
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/RemoveDiskFromConsistencyGroup.java:88]: ditto, throw exception instead

File: compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  public static void cleanUp()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // Delete created consistency group
+    RegionalDelete.deleteRegionalDisk(PROJECT_ID, REGION, DISK_NAME);
+    DeleteDiskConsistencyGroup.deleteDiskConsistencyGroup(
+        PROJECT_ID, REGION, CONSISTENCY_GROUP_NAME);
+  }
+
+  @Test
+  @Order(1)
```
Comment (minherz) [inline compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java:84]: ordering is a bad testing practice. parallel test execution by setting up all necessary resources at setup and clean up either after each test or at cleanup (`@AfterAll`).

Consider implementing mocking like with TPU

**Review by minherz — COMMENTED** [review]
I see that introducing large blocks of comments for alternative implementation is a pattern used in more than one PR (see #9687).

Please remove these commented lines. If you find necessary to demonstrate the same example made for the zonal location disk, please create a stand alone code samples(s).

**Review by minherz — COMMENTED** [review]
I see that introducing large blocks of comments for alternative implementation is a pattern used in more than one PR (see #9687).

Please remove these commented lines. If you find necessary to demonstrate the same example made for the zonal location disk, please create a stand alone code samples(s).

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/RemoveDiskFromConsistencyGroup.java (hunk 5 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/RemoveDiskFromConsistencyGroup.java:88]: Thank you! fixed

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/RemoveDiskFromConsistencyGroup.java (hunk 4 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/RemoveDiskFromConsistencyGroup.java:71]: We are asked to implement all the logic in one sample. Created a check for zonal/regional location

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java (hunk 2 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java:52]: Thank you! Fixed

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java (hunk 3 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java:56]: Thank you! Fixed

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/CreateDiskConsistencyGroup.java (hunk 1 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/CreateDiskConsistencyGroup.java:62]: Thank you!Fixed

File: compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java (hunk 6 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java]: Thank you! Fixed test

**Pull request merged**